### ISSUE TITLE
Implement Natural Language Service

### ISSUE THREAD

**Issue by blowmage** [GoogleCloudPlatform/google-cloud-ruby#828]
Implement the Natural Language service. The implementation should use the GAX middleware. The Ruby API should be similar to Vision, by allowing multiple features to be run at once, or individual features to be run on a Document object. 

```ruby
gcloud = Google::Cloud.new
language = gcloud.language

# Run annotations on a Document object
doc = language.document "Blah blah blah", language: :en
# Run just syntax annotation
syntax = doc.syntax
# Run just entities annotation
entities = doc.entities
# Run just sentiment annotation
sentiment = doc.sentiment

# Run all annotations (syntax, entities, sentiment) in a single call
annotations = language.annotate doc, encoding: :utf8

# Run only syntax annotation, passing in content instead of Document
annotations = language.annotate "Blah <b>blah</b> blah", syntax: true, format: :html

# Run just entities and sentiment annotations for Storage File
file = gcloud.storage.bucket("my_bucket").file("content.html")
annotations = language.annotate file, entities: true, sentiment: true
entities = annotations.entities
sentiment = annotations.sentiment
```

Here are some remaining tasks, not all are needed for release.

- [x] Documentation pass!
- [ ] Enable retries configuration
- [ ] Check content string encoding, possibly auto-convert if not a supported encoding?
- [ ] Validate `encoding` optional argument is one of the supported enum values
- [ ] Add `Token#part_of_speech` helpers?
- [ ] Add `Token#label` helpers?

**Comment by blowmage** [comment]
I think we may have to release without the retry configuration settled. See #848 and #849.

**Comment by blowmage** [comment]
This was released Friday in v0.20.0
